app/src/services/user.js

```diff
@@ -134,18 +149,26 @@ const service = {
    * @returns {Promise<object>} The result of running the login operation
    */
   login: async (token) => {
+    let response;
     const newUser = service._tokenToUser(token);
-    const oldUser = await User.query()
-      .where('identityId', newUser.identityId)
-      .first();
+    // const newUser = { ...service._tokenToUser(token), identityId: 'xyz' };
 
-    if (!oldUser) {
-      // Add user to system
-      return service.createUser(newUser);
-    } else {
-      // Update user data if necessary
-      return service.updateUser(oldUser.userId, newUser);
-    }
+    // wrap with db transaction
+    await utils.trxWrapper(async (trx) => {
+      // check if user exists in db
+      const oldUser = await User.query(trx)
+        .where({ 'identityId': newUser.identityId, idp: newUser.idp })
+        .first();
+        
+      if (!oldUser) {
+        // Add user to system
+        response = await service.createUser(newUser, trx);
+      } else {
+        // Update user data if necessary
+        response = await service.updateUser(oldUser.userId, newUser, trx);
+      }
+    });
+    return response;
```

Comment:
Since we're using the trxWrapper, we should be taking advantage of returning the callback and returning the result of the trxWrapper instead of jumping scope blocks for our response. Ideally we'd want to do something like return await utils.trxWrapper(...) and the callback would return the service.create/update response instead of assigning it to a let value outside its scope.